src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -56,8 +69,32 @@ bool State::operator==(const State& aState) const
         return false;
     }
 
-    return (this->instant_ == aState.instant_) && (this->frameSPtr_ == aState.frameSPtr_) &&
-           (this->coordinates_ == aState.coordinates_);
+    if (this->instant_ != aState.instant_)
+    {
+        return false;
+    }
+
+    if (this->frameSPtr_ != aState.frameSPtr_)
+    {
+        return false;
+    }
```

Comment:
do you mean converting the state from one frame to another regardless and then compare them?

I am not sure, I guess it depends on what this operator is used for.

Reference frame conversion will likely yield similar but not identical values for the coordinates, so even comparing I doubt it will actually be identical (in the sense that the == operator means)

Reply:
> Reference frame conversion will likely yield similar but not identical values for the coordinates, so even comparing I doubt it will actually be identical (in the sense that the == operator means)

Yeah fair point. We can revisit later